src/commands/run.rs

```diff
@@ -376,13 +391,54 @@ impl RunCommand {
         // Invoke the function and then afterwards print all the results that came
         // out, if there are any.
         let mut results = vec![Val::null(); ty.results().len()];
-        func.call(store, &values, &mut results).with_context(|| {
+        let invoke_res = func.call(store, &values, &mut results).with_context(|| {
             if let Some(name) = name {
                 format!("failed to invoke `{}`", name)
             } else {
                 format!("failed to invoke command default")
             }
-        })?;
+        });
+        match invoke_res {
+            Ok(_) => {}
+            Err(err) => {
+                if let Some(coredump_path) = self.coredump_on_trap.as_ref() {
+                    let bt = err
+                        .downcast_ref::<WasmBacktrace>()
+                        .ok_or_else(|| anyhow!("failed to downcast to WasmBacktrace"))?;
```

Comment:
This'll want to be a fallible conversion, only emitting a backtrace in the core dump if `WasmBacktrace` is available I think. (as opposed to returning a different kind of error if the backtrace is missing)

Reply:
Good point. Same with the errors bellow, I'll change the code to allow coredump generation to fail gracefully.